Wrappers/Python/ccpi/optimisation/functions/BlockFunction.py

```diff
@@ -135,27 +130,68 @@ def proximal(self, x, tau, out = None):
             
             
     
-    def gradient(self,x, out=None):
+    def gradient(self, x, out=None):
+        
+        r"""Returns the value of the gradient of the BlockFunction function at x.
         
-        r'''Evaluates gradient of BlockFunction at x
+        .. math:: F'(x) = [f_{1}'(x_{1}), ... , f_{m}'(x_{m})]        
         
-            returns: BlockDataContainer .. math:: [f_{1}'(x_{1}), ... , f_{n}'(x_{n})]
+        Parameter:
+            
+            x : BlockDataContainer and must have as many rows as self.length
                 
-        '''
+        """        
+        
+        if self.length != x.shape[0]:
+            raise ValueError('BlockFunction and BlockDataContainer have incompatible size')        
         
         out = [None]*self.length
         for i in range(self.length):
             out[i] = self.functions[i].gradient(x.get_item(i))
             
-        return  BlockDataContainer(*out)           
+        return  BlockDataContainer(*out)     
+
+    def proximal_conjugate(self, x, tau, out = None):
```

Comment:
Although, `BlockFunction` inherits from `Function`, we need to have `proximal_conjugate` defined in the `BlockFunction` explicitly. Same was used in the `ScaledFunction` [here](https://github.com/vais-ral/CCPi-Framework/blob/merge_master_into_Functions/Wrappers/Python/ccpi/optimisation/functions/Function.py#L297)